### PR TITLE
Bulk invite team members (setup section II)

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

Builds up on #3129 to introduce the last step of setup's Section II.
- Fixes an important security bug in which any user, whether they belonged to an organization or not, could invite members to a given organization ([additional context](https://posthog.slack.com/archives/C0113360FFV/p1612193182042800)).
- Updates the serializer (and types) for `OrganizationInvite` to use the `UserSerializer` instead of fetching every property individually. Plus, some cleanup to the serializer.
- Introduces the ability (API & front-end) to bulk invite up to 20 team members in a single go.

image
Tagging @jamesefhawkins, for feedback on inviting multiple team members on onboarding experience.

## Checklist

- [X] All querysets/queries filter by Organization, by Team, and by User
- [X] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Comment by paolodamico** [comment]
FYI @Twixes too ready for review.

File: frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx
Comment (Twixes) [inline frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx:80]: So as far as I understand, this is to make the experience smoother for the invite. Can the invitee deviate from this optionally preprovided name?

File: posthog/models/organization.py
Comment (Twixes) [inline posthog/models/organization.py:161]: Field name is `first_name`, but is this actually intended for first name explicitly? If this is only a holdover from `User.first_name` (which is kind of a legacy issue), then it should probably be just `name`

**Review by Twixes — COMMENTED** [review]
Just took a quick glance at the code!

**Comment by Twixes** [comment]
The `See suggestion and save` Cy test is again failing 🤔 Not sure if related to this PR

**Comment by paolodamico** [comment]
Thanks for the review @Twixes! As this introduces quite a few changes to the onboarding process I'll wait until tomorrow to merge to see if @macobo has more input.

The not passing test seems like a flaky issue, tests were running before and doesn't look like any relevant front-end code was altered. Running again 👍 

As a bit of feedback on the review if its helpful, I think stuff like ae7fc1e8a15a086abc9142491bcbd1d5d84ebf5e just causes us to spend more time on re-reviews and doesn't really provide a lot of value (purely stylistic).

**Comment by Twixes** [comment]
Oh, I just took a look because of the organization-related fix, but @macobo will definitely have valuable input, especially given previous reviews around this project.

File: frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx
Comment (paolodamico) [reply frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx:80]: Yes, that's the idea! Not yet, but #2734 introduces support for that

**Comment by paolodamico** [comment]
Definitely, super helpful you reviewed that stuff too, thanks! Would've loved to do this in two separate PRs but it ended up being too intertwined to be worth it.

File: posthog/models/organization.py
Comment (paolodamico) [reply posthog/models/organization.py:161]: Well it is indeed to match `User.first_name`, but I thought we were just using first name, not full name everywhere else too (e.g. the signup form says first name and the placeholder says "Jane"). So `first_name` sounds appropriate even then, thoughts?

Comment (Twixes) [reply posthog/models/organization.py:161]: Hmm, I'm pretty sure we're using full names, just using a single field, as we don't use `User.last_name` _anywhere_. 🤔

Comment (Twixes) [reply posthog/models/organization.py:161]: Hm, actually we are kind of using it… accidentally, as social auth fills it in on registration. We should get this straight either way. Thoughts @paolodamico?

Comment (paolodamico) [reply posthog/models/organization.py:161]: Well I'm not sure what most users did before we revamped the signup page, but for recent users not doing social auth, I think it's pretty straightforward to just type your first name. I would just do first name everywhere, it has easier backwards support, it's nicer to receive emails addressed like "Hey John! instead of "Hey John Doe!" and I don't think we really need the last name for anything.

The only reason I can think for this is for very large teams, but in that case maybe the email address is enough, or they could just do full name by their own decision?

File: frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx
Comment (macobo) [inline frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx:40]: Using onBlur seems kind of weird here, not sure this would always work. Why not onChange?

File: frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx
Comment (macobo) [inline frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx:94]: Since we limit this on the backend, limit this in the FE as well?

File: frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts
Comment (macobo) [inline frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts:67]: Nit: I recommend:

```
const payload = {
  invites: [] as { target_email: string | null, first_name: string | null
}
```

Using record only makes sense if the first argument is actually generic not a specific literal.

File: frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts
Comment (macobo) [inline frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts:65]: This seems to be an error - should it be `return { invites: [] }`?

File: frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts
Comment (macobo) [inline frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts:55]: This can be simplified:

```ts
invites.filter((invite) => invite.isValid && !!invite.email).length > 0
```

**Review by macobo — APPROVED** [review]
Some smaller comments again! Otherwise lgtm

File: frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx
Comment (paolodamico) [reply frontend/src/scenes/organization/TeamMembers/BulkInviteModal.tsx:40]: Oops forgot to change that, yes

File: frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts
Comment (paolodamico) [reply frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts:65]: Yes, great catch!

File: frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts
Comment (paolodamico) [reply frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts:55]: Well not quite, there are two things we check here,
a) at least one valid invite **and**,
b) no invalid invites

The proposed code doesn't cover case b).

Comment (macobo) [reply frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts:55]: When is isValid true? Is it enough to check by that?

```
invites.length > 0 && invites.filter(({ isValid }) => !isValid).length == 0
```

Comment (paolodamico) [reply frontend/src/scenes/organization/TeamMembers/bulkInviteLogic.ts:55]: We default to `isValid = true`, let me make some adjustments so we can indeed use the simplified version here.

**Review by macobo — APPROVED** [review]
Did not dig deeply into permissions stuff but solid work :heart:

**Pull request merged**
